app/scripts/modules/core/projects/dashboard/pipeline/ProjectPipeline.tsx

```diff
@@ -34,7 +36,10 @@ export class ProjectPipeline extends React.Component<IProjectPipelineProps, IPro
   private handleStageClick = (stageIndex: number) => $state.go('^.application.pipelines.executionDetails.execution', {application: this.props.execution.application, executionId: this.props.execution.id, stage: stageIndex})
 
   public render() {
-    const stages = this.props.execution.stageSummaries.map((stage) => <ExecutionMarker key={stage.refId} stage={stage} onClick={this.handleStageClick} width={this.state.stageWidth}/>);
+    const application = this.props.application,
+          execution = this.props.execution,
+          stages = execution.stageSummaries.map((stage) =>
+      <ExecutionMarker key={stage.refId} application={application} execution={execution} stage={stage} onClick={this.handleStageClick} width={this.state.stageWidth}/>);
```

Comment:
A little JSX + ES6 magic if you want... since IProps is only application and execution this could change from:
```
   const application = this.props.application,
          execution = this.props.execution,
          stages = execution.stageSummaries.map((stage) =>
   <ExecutionMarker key={stage.refId} application={application} execution={execution} stage={stage} onClick={this.handleStageClick} width={this.state.stageWidth}/>);
```

to:
```
const stages = execution.stageSummaries.map((stage) =>
   <ExecutionMarker key={stage.refId} {...this.props} stage={stage} onClick={this.handleStageClick} width={this.state.stageWidth}/>);